.github/scripts/generate-artifacts.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+set -o errexit
+set -o pipefail
+set -o nounset
+
+repo_root_dir='.'
+deployment_output_folder="$repo_root_dir/broadcast/Deploy.sol"
+
+echo "Creating build artifacts..."
+forge build -o artifacts
+
+echo "Creating networks.json..."
+for deployment in "$deployment_output_folder/"*; do
+  # The subfolder name is the chain id
+  chain_id=${deployment##*/}
+  # First, every single deployment is formatted as if it had its own networks.json
```

Comment:
Could you make a comment on the assumptions about double deployments you are making? They are not allowed at all, right?

Reply:
Added a comment. I'd say the main problems with double deployments (i.e., deploying the same contract twice in the same run of the `Deploy.sol` script) is the `networks.json` format itself more than the execution of this script. Keys in a JSON files aren't supposed to repeat, so we'd have to find a different file format if we really want to capture information about two same-contract deployments. It is tricky however if for some reason we deploy two contracts of the same type but we only care about the first; then we need to change this script.